benchexec/tools/mlb.py

```diff
@@ -0,0 +1,45 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    Tool info for MLB
+    (https://github.com/MLB-SE/Experiment).
+    """
+
+    def executable(self):
+        return util.find_executable("run.sh")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, arg="-version")
+
+    def name(self):
+        return "MLB"
+
+    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+        cmd = [executable]
+        if options:
+            cmd = cmd + options
+        if propertyfile:
+            cmd.append(propertyfile)
+        return cmd + tasks
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        # parse output
+        status = result.RESULT_UNKNOWN
```

Comment:
I would suggest to return `RESULT_UNKNOWN` only if the tool actually gave up itself, and `RESULT_ERROR` in other cases. This will make it easier to distinguish crashes, timeouts, etc. from unknown results, and will allow BenchExec to add more information about the problem in case the tool terminates abnormally.